tests/Integration/Behaviour/Features/Context/Domain/ProductFeatureContext.php

```diff
@@ -55,6 +62,191 @@ public function setProductTaxRulesGroup(string $taxRulesGroupReference, string $
+     *
+     * @param string $productReference
+     */
+    public function assertProductAssignedToDefaultCategory(string $productReference)
+    {
+        $context = $this->getContainer()->get('prestashop.adapter.legacy.context')->getContext();
+        $defaultCategoryId = (int) $context->shop->id_category;
+        $product = $this->getProductByReference($productReference);
+        $productCategories = $product->getCategories();
+
+        $belongsToDefaultCategory = false;
+        foreach ($productCategories as $categoryId) {
+            if ((int) $categoryId === $defaultCategoryId) {
+                $belongsToDefaultCategory = true;
+
+                break;
+            }
+        }
+
+        if ((int) $product->id_category_default !== $defaultCategoryId || !$belongsToDefaultCategory) {
+            throw new RuntimeException('Default category is not assigned to product');
+        }
+    }
+
+    /**
+     * @Then product :productReference type should be :productType
+     *
+     * @param string $productReference
+     * @param string $productTypeName
+     */
+    public function assertProductType(string $productReference, string $productTypeName)
+    {
+        $product = $this->getProductByReference($productReference);
+        $productTypeValue = $this->getProductTypeValueByName($productTypeName);
+
+        $isVirtual = $product->is_virtual;
+        $isPack = Pack::isPack($product->id);
+        $isStandard = !Pack::isPack($product->id) && !$product->is_virtual;
+        $isCombination = !empty($product->getAttributeCombinations());
+
+        if ($isVirtual && $productTypeValue === ProductType::TYPE_VIRTUAL) {
+            return;
+        }
+
+        if ($isPack && $productTypeValue === ProductType::TYPE_PACK) {
+            return;
+        }
+
+        if ($isStandard && $productTypeValue === ProductType::TYPE_STANDARD) {
+            return;
+        }
+
+        if ($isCombination && $productTypeValue === ProductType::TYPE_COMBINATION) {
+            return;
+        }
+
+        throw new RuntimeException(sprintf('Product type is not as expected. Expected %s', $productTypeName));
+    }
+
+    /**
+     * @Then I should get error that product name is invalid
+     */
+    public function assertLastErrorIsReturnProductDisabled()
+    {
+        $this->assertLastErrorIs(
+            ProductConstraintException::class,
+            ProductConstraintException::INVALID_NAME
+        );
+    }
+
+    /**
+     * @Then I should get error that product type is invalid
+     */
+    public function assertLastErrorIsInvalidTypeConstraint()
+    {
+        $this->assertLastErrorIs(
+            ProductConstraintException::class,
+            ProductConstraintException::INVALID_PRODUCT_TYPE
+        );
+    }
+
+    /**
+     * @param string $typeName
+     *
+     * @return int
+     */
+    private function getProductTypeValueByName(string $typeName): int
+    {
+        $typeValueByName = [
```

Comment:
Should be a const as it never changes. Maybe in ProductType class

Reply:
but keywords are specific for a test, technically those strings aren't used anywhere.